iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1644,4 +1647,74 @@ public void validatePartSpec(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
       }
     }
   }
+
+  /**
+   * A function to decide whether a given truncate query can perform a metadata delete or not.
+   * If its not possible to perform metadata delete then try to perform a positional delete.
+   * @param hmsTable A Hive table instance.
+   * @param partitionSpec Map containing partition specification given by user.
+   * @return true if we can perform metadata delete, otherwise false.
+   * @throws SemanticException Exception raised when a partition transform is being used
+   * or when partition column is not present in the table.
+   */
```

Comment:
Thanks for the comment. The method name `shouldTruncate` explains to me that this method is making some decision based on its actual parameters. However the method body does not seems to be self explaining. Could you please extend this javadocs with the steps of the decision making?

Reply:
Done.